e2etests/tests/import_vrf.go

```diff
@@ -0,0 +1,436 @@
+		for _, c := range frrs {
+			err := frrcontainer.PairWithNodes(cs, c, family, func(frr *frrcontainer.FRR) {
+				frr.NeighborConfig.ToAdvertiseV4 = address.FilterForFamily(toAdvertise, ipfamily.IPv4)
+				frr.NeighborConfig.ToAdvertiseV6 = address.FilterForFamily(toAdvertise, ipfamily.IPv6)
+			})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+
+	updateAndCheckPeered := func(config frrk8sv1beta1.FRRConfiguration, peersDefault, peersVRF config.PeersConfig, frrsDefault, frrsVRF []*frrcontainer.FRR, family ipfamily.Family) {
+		err := updater.Update(peersDefault.Secrets, config)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = updater.Update(peersVRF.Secrets, config)
+		Expect(err).NotTo(HaveOccurred())
+
+		nodes, err := k8s.Nodes(cs)
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, c := range frrsDefault {
+			ValidateFRRPeeredWithNodes(nodes, c, family)
+		}
+		for _, c := range frrsVRF {
+			ValidateFRRPeeredWithNodes(nodes, c, family)
+		}
+	}
+
+	baseConfig := frrk8sv1beta1.FRRConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: k8s.FRRK8sNamespace,
+		},
+		Spec: frrk8sv1beta1.FRRConfigurationSpec{
+			BGP: frrk8sv1beta1.BGPConfig{
+				Routers: []frrk8sv1beta1.Router{
+					{
+						ASN: infra.FRRK8sASN,
+					},
+					{
+						ASN: infra.FRRK8sASNVRF,
+						VRF: infra.VRFName,
+					},
+				},
+			},
+		},
+	}
+
+	prefixesToSelectors := func(prefixes []string) []frrk8sv1beta1.PrefixSelector {
+		res := []frrk8sv1beta1.PrefixSelector{}
+		for _, p := range prefixes {
+			selector := frrk8sv1beta1.PrefixSelector{
+				Prefix: p,
+				LE:     32,
+				GE:     0,
+			}
+			ip, _, err := net.ParseCIDR(p)
+			if err != nil {
+				panic(err)
+			}
+			if ip.To4() == nil {
+				selector.LE = 64
+			}
+			res = append(res, selector)
+		}
+		return res
+	}
+
+	ginkgo.Context("while receiving IPs", func() {
+
+		ginkgo.DescribeTable("when the default VRF imports a vrf", func(
+			ipFamily ipfamily.Family,
+			toAdvertise,
+			toAdvertiseVRF []string,
+			allowMode frrk8sv1beta1.AllowMode) {
+
+			frrsDefault, peersDefault, neighborsDefault := initNeighbors(false, ipFamily)
+			frrsVRF, peersVRF, neighborsVRF := initNeighbors(true, ipFamily)
+
+			ginkgo.By("pairing with nodes")
+			pairWithNodes(frrsDefault, ipFamily, toAdvertise)
+			pairWithNodes(frrsVRF, ipFamily, toAdvertiseVRF)
+
+			config := baseConfig.DeepCopy()
+			config.Spec.BGP.Routers[0].Neighbors = neighborsDefault
+			config.Spec.BGP.Routers[0].Imports = []frrk8sv1beta1.Import{{VRF: infra.VRFName}}
+			config.Spec.BGP.Routers[1].Neighbors = neighborsVRF
+
+			for i := range config.Spec.BGP.Routers[0].Neighbors {
+				if allowMode == frrk8sv1beta1.AllowRestricted {
+					config.Spec.BGP.Routers[0].Neighbors[i].ToReceive.Allowed.Prefixes = prefixesToSelectors(append(toAdvertise, toAdvertiseVRF...))
+				}
+				config.Spec.BGP.Routers[0].Neighbors[i].ToReceive.Allowed.Mode = allowMode
+			}
+			for i := range config.Spec.BGP.Routers[1].Neighbors {
```

Comment:
here you use range to get the index, imo the thing should work
```
for _,n:=range xxx{
do something with n
}
```
If it is only to use the array itself a for loop is cleaner.

But the effort to make the first to work (pointers/nested structs etc) might not worth, and therefore keeping as is works for me

Reply:
if "do something with n" is actually changing n, range won't work

Reply:
true because https://github.com/metallb/frr-k8s/blob/main/api/v1beta1/frrconfiguration_types.go#L78 is not an array of pointers, was that a design decision?

(no action needed)

Reply:
why it would be a better design decision for the api?